### DISCUSSION TITLE
Data Importer: unable to (auto)upload JSON only?

### DISCUSSION THREAD

**Discussion by stnokott** [firefly-iii/firefly-iii#5912] [Firefly III] [answered]
Hey there!

I'm running into issues using the POST `/autoupload` endpoint of the otherwise incredible Data Importer (running 0.9.0).

I am executing:
```bash
curl --location --request POST 'http://192.168.178.31:8081/autoupload?secret=verysafesecret' \
--header 'Accept: application/json' \
--form 'json=@"/path/to/local/json"'
```
(`/path/to/local/json` refers to the path to the config JSON on the machine I execute the request on)
(The `Authorization` header mentioned in the sample requests in the [https://docs.firefly-iii.org/data-importer/usage/post/](documentation) did not seem to be necessar with my setup)

**This produces a request response of HTTP 500 with body `{ "message": "Server Error" }`.
The error message on the Data Importer says:**
`production.ERROR: Call to a member function getPathname() on null {"exception":"[object] (Error(code: 0): Call to a member function getPathname() on null at /var/www/html/app/Http/Controllers/AutoUploadController.php:74)`

I tried to find the issue myself, but looking at the actual line in `AutoUploadController.php` that throws the error makes me think that passing a JSON without an accompanying CSV results in `$request->file('csv')` returning null, thus making `$csv->getPathname()` throw the exception:
```php
$json = $request->file('json');
$csv  = $request->file('csv');

try {
    $this->importUpload($csv->getPathname(), $json->getPathname());  // line that produces the error
} catch (ImporterErrorException $e) {
    app('log')->error($e->getMessage());
    $this->line(sprintf('Import exception (see the logs): %s', $e->getMessage()));
}
```
I'm no PHP expert though, that's why I'm asking here instead of creating an issue...

(Using the `/autoimport` endpoint with the config files stored on the machine running the Data Importer works fine btw)

My setup:
- Data Importer in Docker container (port mapping 8080 -> 8081)
- relevant environment variables (mainly `AUTO_IMPORT_SECRET`, `CAN_POST_AUTOIMPORT` & `CAN_POST_FILES` are set)

Could not find any issues/discussions mentioning this error, hope I didn't miss anyone already having answered this question.

Thanks for creating this incredible piece of software, hope someone can help me out here.

**Comment by JC5** [comment, marked as answer]
Nice find, that's a bug. I thought I tested that, but alas.

I hope to release 0.9.1 soon. You can test it already by switching to `:develop`, where the issue should be fixed. Let me know if it works!

**Reply by stnokott** [reply]
Yes, fixed on `:develop` branch, nice work! Looking forward to `0.9.1`!
